Contributing.md link & PR template simplifications & Added ISSUE_TEMPLATE.md

xBalbinus opened the pull request:
<!-- < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < ☺
v                               ✰  Thanks for creating a PR! ✰    
☺ > > > > > > > > > > > > > > > > > > > > > > > > > > > > > > > > >  -->

Closes: #1263 (Not merging until this also gets closed on docs side)

## What is the purpose of the change

- There's a bunch of stuff that if in the current PR template that we can simplify and make smoother.

## Brief change log

- Added a link to the upcoming "contributing.md" located soon on both the docs and this current repo
- Changed current PR template to be smoother (less scrolling for everyone!)

## Tests / Documentation

- No tests altered, documentation soon to be PR'd onto docs.

p0mvn commented on .github/issue_template.md:
```diff
@@ -0,0 +1,17 @@
+*here, you can put concrete steps as to what to do next*
+*(for example)*
+- *create test file in...*
+## Criteria
```
nit: just to be precise that we are referring to a specific concept in Agile: https://www.productplan.com/glossary/acceptance-criteria/
```suggestion
## Acceptance Criteria
```

p0mvn commented on .github/pull_request_template.md:
```diff
@@ -1,52 +1,18 @@
-  - *Daemons retrieve the RPC data from the blob cache*
-
-
-## Testing and Verifying
```
I'd prefer to still have a separate section for testing to remind the contributors and ourselves about its importance

xBalbinus replied:
Got it, readded back

p0mvn replied:
Awesome, thank you.

Thinking more about this, I found the following options useful so far:
```
This change is a trivial rework/code cleanup without any test coverage.

This change added tests and can be verified as follows:
```

I always choose one or the other. Never found myself using the second one. Not sure how others feel, but I would be okay with keeping first and third. Let me know what you think

ValarDragon replied:
yeah, not sure why were deleting this, I thought this section was great

p0mvn reviewed (commented):
Nice, work. I like the issue template. Left a couple of comments too, let me know what you think

ValarDragon commented on .github/pull_request_template.md:
```diff
@@ -1,52 +1,18 @@
-  - *Extended integration test for ...*
-  - *Manually verified the change by ...*
-
-## Documentation and Release Note
-
-  - Does this pull request introduce a new feature or user-facing behavior changes? (yes / no)
-  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? (yes / no)
-  - How is the feature or change documented? (not applicable   /   specification (`x/<module>/spec/`)  /  [Osmosis docs repo](https://github.com/osmosis-labs/docs)   /   not documented)
-
```
I feel like this one was better, it forced acknowledgement of the options for docs updates

ValarDragon commented on .github/pull_request_template.md:
```diff
@@ -1,52 +1,18 @@
-
-
 ## Testing and Verifying
-
-*(Please pick one of the following options)*
-
-This change is a trivial rework / code cleanup without any test coverage.
-
-*(or)*
-
-This change is already covered by existing tests, such as *(please describe tests)*.
-
-*(or)*
-
-This change added tests and can be verified as follows:
-
-*(example:)*
```
Do others dislike this? I thought this was great, I just choose the one I want and delete everything else

ValarDragon commented on .github/pull_request_template.md:
```diff
@@ -1,52 +1,18 @@
-
-## Brief change log
-
+## Brief change log (include any info relevant to tests))
```
what is `(include any info relevant to tests)` for?

alexanderbez replied:
```suggestion
## Brief Changelog
```

xBalbinus commented:
Hmm, got it, depending on additional input I might just revert most if not
all of the quick changes in the pr template…

On Tue, May 3, 2022 at 3:43 PM Dev Ojha ***@***.***> wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In .github/pull_request_template.md
> <https://github.com/osmosis-labs/osmosis/pull/1399#discussion_r864135653>:
>
> > -
> -*(Please pick one of the following options)*
> -
> -This change is a trivial rework / code cleanup without any test coverage.
> -
> -*(or)*
> -
> -This change is already covered by existing tests, such as *(please describe tests)*.
> -
> -*(or)*
> -
> -This change added tests and can be verified as follows:
> -
> -*(example:)*
>
> Do others dislike this? I thought this was great, I just choose the one I
> want and delete everything else
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/osmosis-labs/osmosis/pull/1399#pullrequestreview-960917265>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ASH4FPXBUCYMEK3JGMG77A3VIF6UZANCNFSM5U7VECAQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

alexanderbez commented on README.md:
```diff
@@ -27,6 +27,11 @@ For the most up to date documentation please visit [docs.osmosis.zone](https://d
 
 Thank you for supporting a healthy blockchain network and community by running an Osmosis node!
 
+## Contributing
```
```suggestion
## Contributing

```

alexanderbez commented on .github/pull_request_template.md:
```diff
@@ -1,52 +1,18 @@
-v    If your PR doesn't close an issue, that's OK!  Just remove the Closes: #XXX line!
 ☺ > > > > > > > > > > > > > > > > > > > > > > > > > > > > > > > > >  -->
-
+## Which issue does this close?
 Closes: #XXX
+## Purpose of the change
+*Add a description of the overall background and high level changes that this PR introduces*
```
```suggestion
## Which issue does this close?

closes: #XXX

## Purpose of the change

> Add a description of the overall background and high level changes that this PR introduces*
```

In general, I think these templates should use comments, i.e. `>` as much as possible. What ends up happening is that authors forget or just ignore replacing the section and you end up with a PR that just has useless template text.

xBalbinus commented:
Alright folks! Reverted changes to pull_request_template.md but issue_template.md mostly remains the same. Also fixed @alexanderbez 's comment on README.md. Let me know what you all think :)

alexanderbez commented on .github/pull_request_template.md:
```diff
@@ -9,12 +9,12 @@ Closes: #XXX
 ## What is the purpose of the change
 
-*Add a description of the overall background and high level changes that this PR introduces *
+>Add a description of the overall background and high level changes that this PR introduces
```
```suggestion
> Add a description of the overall background and high level changes that this PR introduces
```

alexanderbez commented on .github/issue_template.md:
```diff
@@ -0,0 +1,17 @@
+<!-- < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < ☺
```
Shouldn't this exist under `.github/ISSUE_TEMPLATE`? Also, what kind of template is this? For bugs? I propose something like the following: https://github.com/umee-network/umee/blob/main/.github/ISSUE_TEMPLATE/bug-report.md

xBalbinus replied:
I didn't happen to know we had a .github/ISSUE_TEMPLATE already if we did (I don't happen to see it), I can definite create a /ISSUE_TEMPLATE folder though. And this was originally supposed to be for generic issues, not specifically bugs.

alexanderbez replied:
Yeah, we should have templates per type IMO (i.e. bugs, general proposal, features, etc...). Otherwise, having a template isn't all that useful.

xBalbinus replied:
Alright, I'll make that then

xBalbinus replied:
I think general proposals / features are done in a separate repo, and questions are done in our discord server, so I currently only implemented the bug reporter template, the generic issue template along with the option to still open blank issues. Let me know your thoughts :)

alexanderbez replied:
I think bug report and general template is fine. I don't really understand why features are done in a separate repo, but I don't wanna bike shed here.

alexanderbez commented on .github/ISSUE_TEMPLATE/issue.md:
```diff
@@ -0,0 +1,18 @@
+v                               ✰  Thanks for creating an issue! ✰    
+☺ > > > > > > > > > > > > > > > > > > > > > > > > > > > > > > > > >  -->
+
+## Background
+*this is where you can provide code examples or context (e.g. past PR's)*
+*(for example)*
+- *The following PRs contributed to...*
+
+## Suggested Design
+*here, you can put concrete steps as to what to do next*
+*(for example)*
+- *create test file in...*
+
+## Acceptance Criteria
+*goals & criteria for success*
+*(for example)*
+- *all existing and new tests should pass*
```
```suggestion
## Background

> This is where you can provide code examples or context (e.g. past PR's)
> e.g.
> - The following PRs contributed to...

## Suggested Design

> Here, you can put concrete steps as to what to do next
> e.g.
> - create test file in...

## Acceptance Criteria

> Goals & criteria for success
> e.g.
> - all existing and new tests should pass
```

p0mvn commented on .github/ISSUE_TEMPLATE/config.yml:
```diff
@@ -0,0 +1 @@
+blank_issues_enabled: true
```
nit: missing new line at the end

xBalbinus commented:
Just wondering, has this been merged in yet?
